UnitTests/TestShared/MockURLSession.swift

```diff
@@ -0,0 +1,14 @@
+import Foundation
+@testable import PaymentsCore
+
+class MockURLSession: URLSessionProtocol {
+    
+    var cannedError: Error?
+    var cannedURLResponse: URLResponse?
+    var cannedData: Data?
```

Comment:
A `String` accessor would allow a [heredoc](https://stackoverflow.com/a/56877844) assignment on the response body to increase test readability. I think it's fair since HTTP is a plain-text protocol.